### DISCUSSION TITLE
freezing

### DISCUSSION THREAD

**Discussion by Gedankenn** [prasanthrangan/hyprdots#216] [Q&A]
I'm having a problem where the entire vm freezes after some time doing nothing. probably i'm the only one having this issue since i'm using it inside a VM. the problem started to occur after some updates, before it i could left it on for weeks. 
If someone has any idea of how i can test it and solve, i would be really grateful.

**Comment by T-Crypt** [comment]
@Gedankenn  

Check out this reddit post -- Hyprland in a VM is not stable and results in a lot of errors / isuses.

https://www.reddit.com/r/archlinux/comments/14hskfa/graphic_bugs_and_hyprland_doesnt_work_on_vm/

**Reply by Gedankenn** [reply]
Ieah, i saw it. i though that maybe because i am running it with passthrough it would behave better. it did for some time. my cpu, gpu and my ssd are with passthrough mode for the vm... i will keep monitoring to see if something changes.

**Reply by T-Crypt** [reply]
@Gedankenn 

Check out the official Hyprland for a few options on the VM to help 

"In a VM, make sure you have 3D acceleration enabled in your virtio config (or virt-manager) otherwise Hyprland will not work.

Please bear in mind 3D accel in VMs may be pretty slow."

https://wiki.hyprland.org/Getting-Started/Master-Tutorial/#vm

**Reply by T-Crypt** [reply]
@Gedankenn 

if you can ssh then please do `sudo gdb attach $(pidof Hyprland)` and provide a `bt -full` to see where it got stuck.

**Reply by T-Crypt** [reply]
gdb probably shows a hang on _malloc_ 

Hyprland or something else could cause a sudden spike in memory use and make you OOM

How much memory do you have on the VM?

**Reply by T-Crypt** [reply]
https://github.com/hyprwm/Hyprland/issues/2789

check out this related issue. (might provide some answers as I don't think its related to the dots files and more Hyprland)
